runtime/cudaq/qis/managers/BasicExecutionManager.h

```diff
@@ -225,14 +228,16 @@ class BasicExecutionManager : public cudaq::ExecutionManager {
 
     if (!adjointQueueStack.empty()) {
       // Add to the adjoint instruction queue
-      adjointQueueStack.back().emplace_back(
-          mutable_name, mutable_params, mutable_controls, mutable_targets, op);
+      adjointQueueStack.back().emplace_back(mutable_name, mutable_params,
```

Comment:
Did we handle the adjoint of custom operations somewhere for non-self-adjoint matrices?
perhaps we need to add `adjoint(unitary)` to the `adjointQueueStack` if it is not self-adjoint here instead.